plutus-ledger/src/Ledger/Scripts.hs

```diff
@@ -84,10 +86,22 @@ import           LedgerBytes                         (LedgerBytes (..))
 -- | A script on the chain. This is an opaque type as far as the chain is concerned.
 newtype Script = Script { unScript :: UPLC.Program UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun () }
   deriving stock Generic
-  deriving Serialise via UPLC.OmitUnitAnnotations UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun
+  deriving newtype Flat
 -- | Don't include unit annotations in the CBOR when serialising.
 -- See Note [Serialising Scripts] in Language.PlutusCore.CBOR
 
+{-| A lot of the plutus infrastructure uses CBOR for serialisation, but we are using
```

Comment:
Specifically: the ledger uses CBOR for transactions, so at some point it has to be CBOR.

Also, this means that the overall CBOR won't be totally "self-describing": it'll just say "here's a bytestring, no idea what's in there" unless you know it's a flat-encoded script. But that's fine.

TBH this is a bit tricky so probably worth a proper Note.

Reply:
I have added a note.